### PR TITLE
Add matching recipe

### PR DESCRIPTION
**Pull request by mat333sp15**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mat333sp15** [comment]
Ack on being a maintainer for this recipe

File: recipes/matching/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 62fa1c2dc1b36f2ea70584d5ca776b3af02e9516140ee4c0c88123c7abe66896
+
+build:
+  number: 0
```
Comment (dopplershift) [inline recipes/matching/meta.yaml:13]: This looks like it's pure Python, so could be noarch python:
```suggestion
  number: 0
  noarch: python
```

File: recipes/matching/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (dopplershift) [inline recipes/matching/meta.yaml:18]: I don't think this actually needs a compiler.
```suggestion
```

File: recipes/matching/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (dopplershift) [inline recipes/matching/meta.yaml:23]: Noarch will need a Python version range here:
```suggestion
    - python >=3.5
    - pip
  run:
    - python >=3.5
```

**Review by dopplershift — APPROVED** [review]

**Pull request merged**